providers/go-feature-flag/src/main/java/org/gofeatureflag/provider/GoFeatureFlagProvider.java

```diff
@@ -0,0 +1,246 @@
+import dev.openfeature.javasdk.exceptions.GeneralError;
+import dev.openfeature.javasdk.exceptions.OpenFeatureError;
+import dev.openfeature.javasdk.exceptions.TypeMismatchError;
+import okhttp3.*;
+import org.gofeatureflag.provider.bean.GoFeatureFlagRequest;
+import org.gofeatureflag.provider.bean.GoFeatureFlagResponse;
+import org.gofeatureflag.provider.bean.GoFeatureFlagUser;
+import org.gofeatureflag.provider.exception.InvalidEndpoint;
+import org.gofeatureflag.provider.exception.InvalidOptions;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static java.net.HttpURLConnection.HTTP_BAD_REQUEST;
+
+public class GoFeatureFlagProvider implements FeatureProvider {
+    private static final String NAME = "GO Feature Flag Provider";
+    private static final ObjectMapper requestMapper = new ObjectMapper();
+    private static final ObjectMapper responseMapper = new ObjectMapper();
+    private final String endpoint;
+    // httpClient is the instance of the OkHttpClient used by the provider
+    private OkHttpClient httpClient;
+
+    public GoFeatureFlagProvider(GoFeatureFlagProviderOptions options) throws InvalidOptions {
+        this.validateInputOptions(options);
+        this.endpoint = options.getEndpoint();
+        this.initializeProvider(options);
+    }
+
+
+    /**
+     * validateInputOptions is validating the different options provided when creating the provider.
+     *
+     * @param options - Options used while creating the provider
+     * @throws InvalidOptions  - if no options are provided
+     * @throws InvalidEndpoint - if the endpoint provided is not valid
+     */
+    private void validateInputOptions(GoFeatureFlagProviderOptions options) throws InvalidEndpoint, InvalidOptions {
+        if (options == null) {
+            throw new InvalidOptions("No options provided");
+        }
+
+        if (options.getEndpoint() == null || "".equals(options.getEndpoint())) {
+            throw new InvalidEndpoint("endpoint is a mandatory field when initializing the provider");
+        }
+    }
+
+    /**
+     * initializeProvider is initializing the different class element used by the provider
+     *
+     * @param options - Options used while creating the provider
+     */
+    private void initializeProvider(GoFeatureFlagProviderOptions options) {
+        // Register JavaTimeModule to be able to deserialized java.time.Instant Object
+        requestMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+        requestMapper.enable(SerializationFeature.INDENT_OUTPUT);
+        requestMapper.registerModule(new JavaTimeModule());
+
+        // init httpClient to call the GO Feature Flag API
+        int timeout = options.getTimeout() == 0 ? 10000 : options.getTimeout();
+        this.httpClient = new OkHttpClient.Builder()
```

Comment:
Does this client use a keepAlive or connection pool?

If you're often hitting the same domain (I guess you are?) you it can improve performance.

Reply:
Yes it always hitting the same domain, I will have proper configuration there you are right.